### PR TITLE
[O2B-1336] Insert default GAQ detector on synchronization with ML

### PR DESCRIPTION
**Pull request by xsalonx**
#### I have a JIRA ticket
- [x] branch and/or PR name(s) include(s) JIRA ID
- [x] issue has "Fix version" assigned
- [x] issue "Status" is set to "In review"
- [x] PR labels are selected

Notable changes for users:
- NA

Notable changes for developers:
- Changed ML synchronization method to set default GAQ detectors for new runs and data passes

Changes made to the database:
- NA

### CODE REVIEW

File: lib/server/services/dataPasses/DataPassService.js [hunk 1]
```diff
@@ -26,6 +26,14 @@ const { BadParameterError } = require('../../errors/BadParameterError');
...
  */
 
+/**
+ * Lists of default GAQ detectors for runs with given pdp_beam_type
+ */
+const DEFAULT_GAQ_DETECTORS_FOR_PDP_BEAM_TYPE = {
+    pp: ['TPC', 'ITS', 'FT0'],
+    PbPb: ['TPC', 'ITS', 'FT0', 'ZDC'],
+};
+
```
Comment (martinboulais) [inline lib/server/services/dataPasses/DataPassService.js:36]: I think that a constant should be an array at max, not something complex like that in general.
In this case, I think that here you should simply make 2 constants and create the object where it's needed

File: lib/server/services/dataPasses/DataPassService.js [hunk 2]
```diff
@@ -75,7 +83,7 @@ class DataPassService {
      * @param {number[]} detectorIds list of ids of DPL detectors included in GAQ computation for the given runs and data pass
      * @return {Promise<void>} promise resolved once association between data pass, runs and detectors was set
      */
-    setGaqDetectors(dataPassId, runNumbers, detectorIds) {
+    async setGaqDetectors(dataPassId, runNumbers, detectorIds) {
```
Comment (martinboulais) [inline lib/server/services/dataPasses/DataPassService.js:86]: No need for async here

File: lib/server/services/dataPasses/DataPassService.js [hunk 3]
```diff
@@ -95,6 +103,50 @@ class DataPassService {
...
+     * Where, for given run, default list of GAQ detectors is intersection of detectors linked to it with list of detectors
+     *  from @see DEFAULT_GAQ_DETECTORS_FOR_PDP_BEAM_TYPE, which depends on beam type of the run
+     *
+     * @param {number} dataPassId data pass id
+     * @param {number[]} runNumbers list of run numbers
+     * @param {object} [options] additional options
+     * @param {sequelize.Transaction} [options.transaction] sequelize transaction
+     * @return {Promise<void>} promise resolved once association between data pass, runs and detectors was set
+     */
+    async setDefaultGaqDetectors(dataPassId, runNumbers, { transaction } = {}) {
```
Comment (martinboulais) [inline lib/server/services/dataPasses/DataPassService.js:117]: I think that the naming here is misleading, I would rather name it `useDefaultGaqDetectors`

**Review by martinboulais — CHANGES_REQUESTED** [review]

File: lib/server/services/dataPasses/DataPassService.js (hunk 3 above)
Comment (xsalonx) [reply lib/server/services/dataPasses/DataPassService.js:117]: applied

File: lib/server/services/dataPasses/DataPassService.js (hunk 2 above)
Comment (xsalonx) [reply lib/server/services/dataPasses/DataPassService.js:86]: applied

File: lib/server/services/dataPasses/DataPassService.js (hunk 1 above)
Comment (xsalonx) [reply lib/server/services/dataPasses/DataPassService.js]: applied

File: lib/database/migrations/20240731134539-add-default-GAQ-detectors.js [hunk 4]
```diff
@@ -38,8 +38,8 @@ module.exports = {
                         AND d.name IN (:detectorsNames)
             `, { transaction, replacements: { beamType, detectorsNames } });
 
-        await setDefaultGaqDetectorsForRunsWithGivenPdpBeamType('pp', ['TPC', 'ITS', 'FT0']);
-        await setDefaultGaqDetectorsForRunsWithGivenPdpBeamType('PbPb', ['TPC', 'ITS', 'FT0', 'ZDC']);
+        await useDefaultGaqDetectorsForRunsWithGivenPdpBeamType('pp', ['TPC', 'ITS', 'FT0']);
```
[the remark this answers is not in the document]
Comment (xsalonx) [reply lib/database/migrations/20240731134539-add-default-GAQ-detectors.js:41]: applied

File: test/lib/server/services/dataPasses/DataPassService.test.js [hunk 5]
```diff
@@ -229,7 +229,7 @@ module.exports = () => {
 
             await dataPass.addRuns(await RunRepository.findAll({ where: { runNumber: { [Op.in]: runNumbers } } }));
 
-            await dataPassService.setDefaultGaqDetectors(dataPassId, runNumbers);
+            await dataPassService.useDefaultGaqDetectors(dataPassId, runNumbers);
 
             expect((await dataPassService.getGaqDetectors(dataPassId, 777770)).map(({ name }) => name)).to
                 .have.all.members(['TPC', 'ITS', 'FT0']);
```
[the remark this answers is not in the document]
Comment (xsalonx) [reply test/lib/server/services/dataPasses/DataPassService.test.js:239]: applied

**Review by martinboulais — APPROVED** [review]

**Pull request merged**